botwo/botwo.py

```diff
@@ -0,0 +1,188 @@
+import fnmatch
+
+SPECIAL_METHODS = {"get_paginator", "delete_objects"}
+COPY_METHODS = {"copy", "copy_object", "copy_upload_part"}
+
+
+def client(client_mapping, config):
+    botwo = Botwo()
+    return botwo.create_client_class(client_mapping, config)
+
+
+class Paginatwo(object):
+
+    def __init__(self, mapping, kwargs, client1, client2, client1_name, client2_name):
+        self.mapping = mapping
+        self.client1_name = client1_name
+        self.client2_name = client2_name
+        self.paginator1 = client1.get_paginator(kwargs["operation_name"])
+        self.paginator2 = client2.get_paginator(kwargs["operation_name"])
+
+    def paginate(self, **kwargs):
+        paginator_to_call = self.paginator1
+        for map in self.mapping:
+            if fnmatch.fnmatch(kwargs["Bucket"], map["bucket_name"]):
+                if map.get("client"):
+                    if map.get("client") == self.client2_name:
+                        paginator_to_call = self.paginator2
+                    elif map.get("client") == self.client1_name:
+                        paginator_to_call = self.paginator1
+                    else:
+                        raise ValueError("Invalid client name " + map.get("client"))
```

Comment:
E.g. if we used the user-supplied dicts directly this would be:
```suggestion
                try:
                    client = self.clients_by_name[map["client"]]
                except KeyError:
                    raise UnknownClientError(map["client"])
```
(I'm assuming we've already validated all map rows during construction, as suggested in the next comment.)

Reply:
Done